config.go

```diff
@@ -130,14 +126,14 @@ func validateConfig(config *Config) error {
 	switch {
 	case config == nil:
 		return errNoConfigProvided
-	case config.Certificate != nil && config.PSK != nil:
+	case config.Certificate.PrivateKey != nil && config.PSK != nil:
```

Comment:
I notice this in a few places, shouldn't we check both Certificate.Certificate and Certificate.PrivateKey?  Maybe write one validation function that's reused every place to make sure both those fields are set properly?

Reply:
I double checked and this should be good! If we are a Client `Certificate.Certificate` may be `nil`.

We do `Config` validation in three different places `Server` `Client` and `validateConfig` I think it would be a great improvement to merge these all into `validateConfig`!